### PR TITLE
Add query DSL module in doc

### PR DESCRIPTION
**Pull request by BossZou**
#270

### CODE REVIEW

File: doc/source/res/query.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
+
+Inspired by ElasticSearch, Milvus provides a Query DSL(Domain Specific Language) consisting of two types of clauses to define queries:
+
+
+**Leaf query clauses**
+
+Leaf query clauses look for a particular value in a particular field. Currently milvus support `term`, `range`, `vector` queries.
+
+  * term: term query matches the entities which corresponding field value are in the specified list. The format is `term: [$value1, $value2, ...]`
```
Comment (XuanYang-cn) [inline doc/source/res/query.md:9]: Grammarly, which should be whose

File: doc/source/res/query.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+
+  * range: range query matches the entities which corresponding field value are in the specified range. The supported range types are:
```
Comment (XuanYang-cn) [inline doc/source/res/query.md:12]: Grammarly, which should be whose

File: doc/source/res/query.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+    - "GT"(greater than)
+    - "GTE"(greater than or equal)
+    - "LT"(less than)
+    - "LTE"(less than or equal).
+
+
+  * vector: vector query only take effect on vector fields, and approximately search nearest vectors. The format should be `{"topk": $topk, "query": $vectors, "params": {...}, "metric_type": $metric}`.
```
Comment (XuanYang-cn) [inline doc/source/res/query.md:20]: Grammarly, `take` should be `takes`

File: doc/source/res/query.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+    here, "topk" is the number of approximately nearest top-k entities for each query vector. "query" is a list of query vectors. "params" is search parameters, and "metric_type" indicates which distance computation type to be used.
+
+Note that, the `term` and `range` query act as filter queries for `vector` query, they are pre-filtered.
+
+**Compound query clauses**
+
+Currently, milvus only support boolean query, i.e. `bool`. There are three occurrence types:
```
Comment (XuanYang-cn) [inline doc/source/res/query.md:28]: Grammarly, `occurrence` means an incident or event, "There are three types" maybe better.

File: doc/source/res/query.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
++--------------------------+---------------------------------------------------------------------+
+| Occurrence types         | Description                                                         |
```
Comment (XuanYang-cn) [inline doc/source/res/query.md:31]: Like the former one.

File: doc/source/res/query.md (hunk 4 above)
Comment (BossZou) [reply doc/source/res/query.md:28]: The name `occurrence` is copied from ES.

Comment (XuanYang-cn) [reply doc/source/res/query.md:28]: Then we found a bug. < - <

Comment (XuanYang-cn) [reply doc/source/res/query.md:28]: I trust google more than ES < - <

File: requirements.txt [hunk 6]
```diff
@@ -23,15 +23,17 @@ Jinja2==2.10.3
...
 lazy-object-proxy==1.4.3
+m2r==0.2.1
 MarkupSafe==1.1.1
 mccabe==0.6.1
+mistune==0.8.4
 mock==3.0.5
 more-itertools==8.1.0
 numpy==1.18.1
 packaging==20.0
+pandas==1.1.3
```
Comment (XuanYang-cn) [inline requirements.txt:34]: What are pandas for??????

Comment (BossZou) [reply requirements.txt:34]: Not necessary, remove it.

File: examples/extract_films.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+
+def random_vector(dim):
+    return [np.float32(random.random()) for _ in range(dim)]
+
+
+def main():
+    out = list()
+
+    with open("/home/yhz/Downloads/ml-latest-small/movies.csv", 'r') as f:
```
Comment (XuanYang-cn) [inline examples/extract_films.py:14]: If `movies.csv` is unavailable, this script is also not useful.  And the file path here...
By the way, I edit thousands of lines for title ends with `, The` of the result from `films.csv`

File: doc/source/res/query.md (hunk 1 above)
Comment (BossZou) [reply doc/source/res/query.md]: changed.

File: doc/source/res/query.md (hunk 2 above)
Comment (BossZou) [reply doc/source/res/query.md]: changed,

**Pull request merged**